activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -237,7 +237,11 @@ def perform_calculation(operation, column_name)
 
         # If #count is used with #distinct (i.e. `relation.distinct.count`) it is
         # considered distinct.
-        distinct = distinct_value
+        if select_values.any? && select_values.first =~ /\s*DISTINCT[\s(]+/i
```

Comment:
Maybe we could simplify this to `distinct = select_values.first&.match?(/\s*DISTINCT[\s(]+/i) || distinct_value`?

As we prefer `match?` to `=~`. Also is it `select_values.first` and not `select_values.any? { |v| v.match?(…)`?